src/attestation_types/quote.rs

```diff
@@ -19,7 +19,24 @@ pub const INTELVID: [u8; 16] = [
     0x93, 0x9A, 0x72, 0x33, 0xF7, 0x9C, 0x4C, 0xA9, 0x94, 0x0A, 0x0D, 0xB3, 0x95, 0x7F, 0x06, 0x07,
 ];
 
+#[derive(Debug, Clone)]
+/// Error type for Quote module
+pub struct QuoteError(String);
+
+impl fmt::Display for QuoteError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{:?}", &self.0)
+    }
+}
+
+impl std::error::Error for QuoteError {
+    fn description(&self) -> &str {
+        &self.0
+    }
+}
```

Comment:
```suggestion
impl std::error::Error for QuoteError {}
```

The [description](https://doc.rust-lang.org/std/error/trait.Error.html#method.description) method is deprecated. I think as long as `QuoteError` is `Debug` and `Display`, it can be `Error` without having to impl any of the trait methods since they have a default implementation. Callers will pass the error object into `println!()` directly to take advantage of the `Display` trait instead of calling `description`

Reply:
@connorkuehl Oh, thanks for pointing that out, I hadn't seen that. It wouldn't let me derive `Display`, so I think that may be implicit also... but seems to compile fine with just `Debug`.

Reply:
@lkatalin I think you still want `impl std::error::Error for QuoteError {}`, otherwise it could be awkward for client code trying to propagate errors down as they'll have to wrap the `QuoteError` struct with something that `impl std::error::Error`. I don't see it in the latest revision. I think it should still work with the empty braces.

Reply:
Okay, this version should be better.